sys/net/routing/rpl/rpl_control_messages.c

```diff
@@ -386,13 +387,13 @@ void rpl_send_DAO(rpl_dodag_t *my_dodag, ipv6_addr_t *destination, uint8_t lifet
     uint8_t entries = 0;
     uint8_t continue_index = 0;
 
-    for (uint8_t i = start_index; i < rpl_max_routing_entries; i++) {
-        if (rpl_get_routing_table()[i].used) {
+    for (uint8_t i = start_index; i < RPL_MAX_PARENTS; i++) {
+        if(rpl_parents[i].used) {
             rpl_send_opt_target_buf->type = RPL_OPT_TARGET;
             rpl_send_opt_target_buf->length = RPL_OPT_TARGET_LEN;
             rpl_send_opt_target_buf->flags = 0x00;
             rpl_send_opt_target_buf->prefix_length = RPL_DODAG_ID_LEN;
-            memcpy(&rpl_send_opt_target_buf->target, &rpl_get_routing_table()[i].address,
+            memcpy(&rpl_send_opt_target_buf->target, &rpl_parents[i].addr,
```

Comment:
Technically yes, but I think this is not exactly what we want since the FIB contains probably more destinations then RPL has in use, other DODAGs, entries from other routing protocols, static routes etc.
That tricky.